packages/core/src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+
+const useLocalStorage = (
+    key: string,
+    initialValue: any,
+    expiryInDays?: number | null,
+) => {
+    const [storedValue, setStoredValue] = useState(() => {
+        if (typeof window === 'undefined' || !window?.localStorage) {
+            return initialValue;
+        }
+        try {
+            const item = window.localStorage.getItem(key);
+            if (item) {
+                const parsedItem = JSON.parse(item);
+                //set the value only if it is not expired
+                if (
+                    new Date().getTime() < new Date(parsedItem.expiry).getTime()
+                ) {
+                    return parsedItem.value;
+                } else {
+                    // remove the value if it is expired
+                    window.localStorage.removeItem(key);
+                }
+            }
+            return initialValue;
+        } catch (error) {
+            console.log(error);
+            return initialValue;
+        }
+    });
+
+    const setValue = (value: any) => {
+        try {
+            let expiry;
+            if (expiryInDays) {
```

Comment:
Yeah makes sense, I have changed the default value of this prop to `null`  so it will behave as a state now.
And if that prop value is given then it would use timestamp logic..

should I remove it completely? 🤔